File: behavioral/strategy/java/src/UPIPaymentStrategy.java
public class UPIPaymentStrategy implements PaymentStrategy {
  private String upiId;

  public UPIPaymentStrategy(String upiId) {
    this.upiId = upiId;
  }

  @Override
  public boolean pay(double amount) {
    System.out.println("Paying " + amount + " via UPI with ID " + upiId);
    return true;
  }
}

File: behavioral/strategy/ruby/lib/upi_payment_strategy.rb
require_relative 'payment_strategy'

class UpiPaymentStrategy < PaymentStrategy
  attr_reader :upi_id

  def initialize(upi_id)
    @upi_id = upi_id
  end

  # Overrides PaymentStrategy#pay
  def pay(amount)
    puts "Paying #{amount} via UPI with ID #{upi_id}"
    true
  end
end
